### PR TITLE
add decorator to person

### PR DESCRIPTION
**Pull request by Daniel-Kigozi**
In this milestone, I did the following
- Add class Nameable
- Add decorators

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,9 +1,12 @@
+require './nameable'
+
 # Represents a person with basic information such as name, age, and permission to use services.
-class Person
+class Person < Nameable
   attr_reader :id
   attr_accessor :name, :age
 
   def initialize(age, parent_permission: true, name: 'Unknown')
+    super(name: name)
```
Comment (codecaiine) [inline person.rb]: - [ ] Good job but unfortunately when I run your app, I got the error message below ⬇️  about the wrong number of arguments in the initialize method 😢 . Please fix it and make your app works as expected and make sure that Trimmer needs to return this `Maximilan ` 👍 .

decorate

**Review by codecaiine — CHANGES_REQUESTED** [review]
Hi @Daniel-Kigozi  , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Good decorate class setup ✔️ 
- No rubocop errors ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 2]
```diff
@@ -1,9 +1,12 @@
+require './nameable'
+
 # Represents a person with basic information such as name, age, and permission to use services.
-class Person
+class Person < Nameable
   attr_reader :id
   attr_accessor :name, :age
 
   def initialize(age, parent_permission: true, name: 'Unknown')
+    super(name)
```
Comment (TedLivist) [inline person.rb]: - [ ] I had the same error as the previous reviewer. Kindly remove `name` from this `#super` method. It should be like so:

  ```
  super()
  ```

File: decorator.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+require './nameable'
+
+# Represents base decorator
+class Decorator < Nameable
+  def initialize(nameable)
+    super(name)
```
Comment (TedLivist) [inline decorator.rb]: - [ ] Kindly remove `name` here also.

  ```
  super()
  ```

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Daniel-Kigozi,

✔️ Implementation of trimmer class

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KayLemba — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hello @Daniel-Kigozi        , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉  Keep up the good work 🥂 

HIGHLIGHTS:

- [x]  Professional README
- [x] No linter errors
- [x] Project correctly setup
- [x] Decorator runs as intended
- [x] Descriptive P.R


 
 

 










 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]]

**Pull request merged**